### ISSUE TITLE
entity.identifier.resolvable

### ISSUE THREAD

**Issue by gothub** [NCEAS/metadig-checks#78]
## Description

For each data entity, check that it's identifier is resolvable

## Priority

- FAIR: Required

## Issues
- should every entity identifier be checked, or just one 'representative' one, as there may be hundreds of entities in a dataset?
- should identifiers be fully resolvable, i.e. should the identifier itself be a resolvable URI, or is it reasonable to use a ‘bare’ identifier and use the DataONE services to resolve it. 
- note that many identifiers for resources are not resolvable in DataONE, for example PANGAEA, which uses ids from their own external repository, for data resources. 
- also note that EDI identifiers are fully resolvable.
- the DataONE 'resolve' check is dependent on a new release of DataONE CCI (as of 2019-07-09)
   - see https://redmine.dataone.org/issues/8812#change-31446

## Procedure

- retrieve the first entity identifier from the metadata (should we check all ids?)
- if the identifier is a URL, i.e. starts with "http.*://", then try to resolve it directly
  - otherwise, check if the id is resolvable, using the DataONE `resolve` service (urlencode the id first)
- if ids are not present and resolvable, the check fails and the non-resolvable id are listed

**Comment by gothub** [comment]
Note that this check is currently disabled in the FAIR suite, until  https://redmine.dataone.org/issues/881 is closed and the fix is incorporated into a DataONE CCI release.

**Comment by gothub** [comment]
This check is now enabled in the FAIR suite since commit 3a8241a07388da87fddb210554a92eb901fa8024

**Comment by gothub** [comment]
Remove invalide XPath `/eml/@id`which is not an entity identifier. This leaves the XPath as
```
          /*/identificationInfo/*/citation/CI_Citation/identifier/MD_Identifier/code//text()[normalize-space()] |
          /*/identificationInfo/*/citation/CI_Citation/identifier/RS_Identifier/code//text()[normalize-space()] |
          /*/identifier |
          /eml/dataset/*[self::dataTable|self::spatialRaster|self::spatialVector|self::storedProcedure|self::view|self::otherEntity]/@id |
          /eml/dataset/*/alternateIdentifier |
          /resource/identifier[identifierType='DOI'] |
          /resource/alternateIdentifiers/alternateIdentifier
```

Fixed in commit a9094da62f0d60da4d8bcb63a0e0639861000f76
